lib/AST/Type.cpp

```diff
@@ -1544,33 +1544,41 @@ bool TypeBase::isExactSuperclassOf(Type ty) {
   return false;
 }
 
-/// Returns true if type `a` has archetypes that can be bound to form `b`.
-bool TypeBase::isBindableTo(Type b) {
-  class IsBindableVisitor : public TypeVisitor<IsBindableVisitor, bool, CanType>
-  {
-    llvm::DenseMap<ArchetypeType *, CanType> Bindings;
-
-  public:
-    IsBindableVisitor() {}
+namespace {
+class IsBindableVisitor
+  : public TypeVisitor<IsBindableVisitor, CanType, CanType>
```

Comment:
Would you mind splitting the extraction change out from the semantics changes?  There's a lot to pick apart here.

Reply:
OK, I'll try to pull out more separate commits. I might see if I'm able to completely eradicate `getCurGenericContext` while I'm at it.